### PR TITLE
doxygen: add v1.10.0, improve maintainability

### PR DESCRIPTION
**Pull request by valgur**

### CODE REVIEW

**Comment by tkhyn** [comment]
Hi,

Doxygen < 1.9.8 cannot be built with CMake >= 3.27 because of [CMP0148](https://cmake.org/cmake/help/latest/policy/CMP0148.html). Could a review of this PR be prioritized?

I also have some improvement suggestions in [this commit](https://github.com/conan-io/conan-center-index/commit/fce8dcd7536f9f8eb0e8b3b5ed777707fe286284). As there are already 2 PRs on this recipe I will refrain from opening a new PR unless requested to. This commit:

- adds CPython as a build requirement (in case Python is not available globally on the build system. Edge case but definitely possible as it happened to me on a new machine)
- package_info made a bit more Conan2-friendly (i.e. properly adding the bin directory to runenv and buildenv)

File: recipes/doxygen/all/conanfile.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 from conan import ConanFile
 from conan.errors import ConanInvalidConfiguration
 from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
-from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, replace_in_file
```
Comment (uilianries) [inline recipes/doxygen/all/conanfile.py:4]:
```suggestion
from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get
```

**Comment by uilianries** [comment]
Just generated the missing package for `xapian-core`: https://c3i.jfrog.io/c3i/misc-v2/summary.html?json=https://c3i.jfrog.io/c3i/misc-v2/logs/prod-v2/TapaholesRepo/7/summary.json

Restarted the build now.

**Comment by gvol** [comment]
I would love a newer version of doxygen in conan so that I can use it at work.  There are some nice features I'd like to use.  

I'm not very good with conan, but is there anything I can do to help?

**Comment by valgur** [comment]
Resolved by #25928.

**Pull request closed without merging**